docs/source/design/clis.rst

```diff
@@ -22,14 +22,54 @@ The client code is located in ``flytekit/clients`` and there are two.
+The ``pyflyte package`` command parses and compiles the user’s Python code into Flyte protobuf objects.
+The ``flytectl register`` command ships the protobuf objects over the network to Flyte control plane, wherein ``flytectl`` also allows adding attributes such as IAM role, K8s service account, and so on. 
+
+Why should you use ``pyflyte register``?
+========================================
+The ``pyflyte register`` command bridges the gap between ``pyflyte package`` + ``flytectl register``and ``pyflyte run`` commands. It offers the functionality of the ``pyflyte package`` (with smarter naming semantics and combines the network call into one step), by registering all the workflows in the repository, and the convenience of the ``pyflyte run``, by launching the execution of a single workflow. 
+
+.. note ::
+
+   You can’t use ``pyflyte register`` if you are unaware of the run-time options yet(IAM role, service account, and so on). 
+
+What is ``pyflyte run``? 
+========================
+The ``pyflyte run`` command is a light-weight, convenience command that combines packaging, registering and launching a workflow in a single command.
+It is not a fully featured production scale mode of operation, because it
+is designed to be a quick and easy iteration tool to get started with Flyte or test small self-contained scripts. The caveat here is it operates on a single file, and this file will have to contain all the required Flyte entities. Let’s take an example so that you can understand it better. 
+Suppose you execute a script that defines 10 tasks and a workflow that calls only 2 out of the 10 tasks, the remaining 8 tasks don’t get registered at that point. 
+It is considered fast registration because when a script is executed using ``pyflyte run``, the script is zipped up and sent to FlyteAdmin. When the task is executed in the backend, this zipped file is extracted and used.
+
+Usage
+=====
+
+.. prompt:: bash $
+
+   pyflyte --config ~/.flyte/dev-uniondemo.yaml register --image ghcr.io/flyteorg/flytecookbook:core-latest --image trainer=ghcr.io/flyteorg/flytecookbook:core-latest --image predictor=ghcr.io/flyteorg/flytecookbook:core-latest --raw-data-prefix s3://development-service-flyte/reltsts flyte_basics
+
+.. note ::
+
+   Neither ``pyflyte register`` nor ``pyflyte run`` commands work on Python namespace packages since both the tools traverse the filesystem to find the first folder that doesn't have an __init__.py file, which is interpreted as the root of the project. Both the commands use this root as the basis to name the Flyte entities.
+
+In a broad way, ``pyflyte register`` is equivalent to ``pyflyte run`` minus launching workflow with the exception that ``pyflyte run`` can only register a single workflow whereas ``pyflyte register`` can register all workflows in a repository.
+
+Should I use ``pyflyte run`` or ``pyflyte package`` + ``flytectl register``?
```

Comment:
Can we also differentiate between "pyflyte package + flytectl register" and just "pyflyte register"?

To me there is a difference.  The first option will produce a "portable" package of your flyte entities.  that is you can take that .tgz file, and register it with ten different flyte backends.  you can email it to your friend at a different company and ask them to register it with their admin.  you can also save the asset, keep it around for audit, etc. 
 pyflyte register works well for single admin use-cases, and the compiled objects are lost afterwards, so you can't as easily introspect.  In a production Flyte setting, I think each has their place.  If I were running something in CI, I would opt for the first route.  If i were a user iterating locally, I'd probably opt for the second way.  @kumare3 thoughts?